packages/botonic-nlp/src/dataset/input-parser.ts

```diff
@@ -0,0 +1,88 @@
+import { lstatSync, readdirSync, readFileSync } from 'fs'
+import { load as loadYaml } from 'js-yaml'
+import { extname, join } from 'path'
+
+import { unique } from '../utils/array-utils'
+import { AugmenterMap, DataAugmenter } from './data-augmenter'
+import { DefinedEntity, EntitiesParser } from './entities-parser'
+
+type InputData = {
+  class?: string
+  entities?: string[]
+  'data-augmentation'?: AugmenterMap
+  samples: string[]
+}
+
+export type Sample = { text: string; class: string; entities: DefinedEntity[] }
+
+type ParsedData = { classes: string[]; entities: string[]; samples: Sample[] }
+
+export class InputParser {
+  public readonly ALLOWED_EXTENSIONS = ['.yaml', '.yml']
+
+  constructor(private path: string) {}
+
+  parse(): ParsedData {
+    const files = this.getInputFiles()
+
+    let classes: string[] = []
+    let entities: string[] = []
+    let samples: Sample[] = []
+
+    files.forEach(filePath => {
+      const parsedData = this.parseInputFile(filePath)
+      classes = classes.concat(parsedData.classes)
+      entities = entities.concat(parsedData.entities)
+      samples = samples.concat(parsedData.samples)
```

Comment:
Since unique will create a Set, It's more efficient to directly use a Set (and to Array.from in the return).
efficiently is specially important here if we have many samples to avoid using too much memory

Reply:
Since Set doesn't have a `concat` method, it would look something like:

```typescript
parse(): ParsedData {
  const files = this.getInputFiles()

  const classes: Set<string> = new Set()
  const entities: Set<string> = new Set()
  const samples: Set<Sample> = new Set()

  files.forEach(filePath => {
    const parsedData = this.parseInputFile(filePath)
    parsedData.classes.forEach(c => classes.add(c))
    parsedData.entities.forEach(e => entities.add(e))
    parsedData.samples.forEach(s => samples.add(s))
  })

  return {
    classes: Array.from(classes),
    entities: Array.from(entities),
    samples: Array.from(samples),
  }
}
```

Do you think this is better? We are also itterating over each element of the parsedData. @dpinol

Reply:
yes, that's better in terms of memory efficiency (which is only important when dealing with very big data, such as samples).
in this case, 
* we avoid keeping in the array in memory multiple copies of repeated samples
* we avoid having the data in 3 places (array with repeated, set and unique array)